pkg/operator/controllers/pullsecret/pullsecret_controller_test.go

```diff
@@ -163,12 +286,532 @@ func TestPullSecretReconciler(t *testing.T) {
+			for _, wantKey := range tt.wantKeys {
+				for _, outKey := range out {
+					if outKey == wantKey {
+						continue KEYS
+					}
+				}
+				t.Fatalf("Cannot find key: %s in condition keys", wantKey)
+			}
+		})
+	}
+}
+
+func TestKeyCondition(t *testing.T) {
+	test := []struct {
+		name          string
+		failed        bool
+		keys          []string
+		wantCondition status.Condition
+		wantErr       string
+	}{
+		{
+			name:   "cannot parse keys",
+			failed: true,
+			keys:   []string{},
+			wantCondition: status.Condition{
+				Type:    arov1alpha1.RedHatKeyPresent,
+				Status:  corev1.ConditionFalse,
+				Message: "Cannot parse pull-secret",
+				Reason:  "CheckFailed",
+			},
+		},
+		{
+			name: "no key found",
+			keys: []string{},
+			wantCondition: status.Condition{
+				Type:    arov1alpha1.RedHatKeyPresent,
+				Status:  corev1.ConditionFalse,
+				Message: "No Red Hat key found in pull-secret",
+				Reason:  "CheckDone",
+			},
+		},
+		{
+			name: "keys found",
+			keys: []string{"registry.redhat.io"},
+			wantCondition: status.Condition{
+				Type:    arov1alpha1.RedHatKeyPresent,
+				Status:  corev1.ConditionTrue,
+				Message: "Red Hat keys found: registry.redhat.io,",
+				Reason:  "CheckDone",
+			},
+		},
+	}
+
+	for _, tt := range test {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &PullSecretReconciler{}
+
+			out := r.keyCondition(tt.failed, tt.keys)
+			if !reflect.DeepEqual(out, &tt.wantCondition) {
+				t.Fatalf("Condition does not match. want: %v, got: %v", tt.wantCondition, out)
+			}
+		})
+	}
+}
+
+func TestBuildRedHatKeyCondition(t *testing.T) {
+	test := []struct {
+		name          string
+		secret        corev1.Secret
+		wantCondition status.Condition
+	}{
+		{
+			name: "RH pull secret is present",
+			secret: corev1.Secret{
+				Data: map[string][]byte{
+					corev1.DockerConfigJsonKey: []byte(`{"auths":{"arosvc.azurecr.io":{"auth":"ZnJlZDplbnRlcg=="},"registry.redhat.io":{"auth":"ZnJlZDplbnRlcg=="}}}`),
+				},
+			},
+			wantCondition: status.Condition{
+				Type:    arov1alpha1.RedHatKeyPresent,
+				Status:  corev1.ConditionTrue,
+				Reason:  "CheckDone",
+				Message: "Red Hat keys found: registry.redhat.io,",
+			},
+		},
+		{
+			name: "RH pull secret is missing",
+			secret: corev1.Secret{
+				Data: map[string][]byte{
+					corev1.DockerConfigJsonKey: []byte(`{"auths":{"arosvc.azurecr.io":{"auth":"ZnJlZDplbnRlcg=="}}}`),
+				},
+			},
+			wantCondition: status.Condition{
+				Type:    arov1alpha1.RedHatKeyPresent,
+				Status:  corev1.ConditionFalse,
+				Reason:  "CheckDone",
+				Message: "No Red Hat key found in pull-secret",
+			},
+		},
+	}
+
+	for _, tt := range test {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &PullSecretReconciler{}
+
+			out, err := r.buildRedHatKeyCondition(&tt.secret)
+			if err != nil {
+				t.Fatalf("Unexpected error")
+			}
+			if !reflect.DeepEqual(*out, tt.wantCondition) {
+				t.Fatalf("Condition does not match: \n%v\n%v", out, tt.wantCondition)
```

Comment:
When printing you might want to print same "format" as you compare, otherwise it confuses reader as you will see
```
&{RedHatKeyPresent ...
{RedHatKeyPresent ...
```
and you might think its pointer vs non-pointer issue from the output. But this is not true.
+ I would notice this if this test would not be failing CI for other reasons :)